src/NSwag.CodeGeneration.CSharp/Templates/ClientTemplate.tt

```diff
@@ -126,44 +126,43 @@ public partial interface I<#=Model.Class#>
 <#  }else if(parameter.IsArray){#>
-        url_ = url_.Replace("{<#=parameter.Name#>}", System.Uri.EscapeDataString(string.Join(",", <#=parameter.VariableName#>.Select(s_ => s_.ToString()))));
+        urlBuilder_.Replace("{<#=parameter.Name#>}", System.Uri.EscapeDataString(string.Join(",", <#=parameter.VariableName#>.Select(s_ => s_.ToString()))));
 <#  }else{#>
-        url_ = url_.Replace("{<#=parameter.Name#>}", System.Uri.EscapeDataString(<#=parameter.VariableName#>.ToString()));
+        urlBuilder_.Replace("{<#=parameter.Name#>}", System.Uri.EscapeDataString(<#=parameter.VariableName#>.ToString()));
 <#  }#>
 <#}#>
 <#foreach(var parameter in operation.QueryParameters){#>
-
-<#  if(!parameter.IsNullable){#>
-        if (<#=parameter.VariableName#> == null)
-            throw new System.ArgumentNullException("<#=parameter.VariableName#>");
-        else
-<#  }else{#>
-        if (<#=parameter.VariableName#> != null)
-<#  }#>
+        <#  if(parameter.IsNullable){#>if (<#=parameter.VariableName#> != null) <#}#>
 <#  if(parameter.IsDateArray){#>
-            foreach (var item_ in <#=parameter.VariableName#>) { url_ += string.Format("<#=parameter.Name#>={0}&", System.Uri.EscapeDataString(item_.ToString("s", System.Globalization.CultureInfo.InvariantCulture))); }
+foreach (var item_ in <#=parameter.VariableName#>) { urlBuilder_.Append("<#=parameter.Name#>=").Append(System.Uri.EscapeDataString(item_.ToString("s", System.Globalization.CultureInfo.InvariantCulture))).Append("&"); }
 <#  }else if(parameter.IsDate){#>
-            url_ += string.Format("<#=parameter.Name#>={0}&", System.Uri.EscapeDataString(<#=parameter.VariableName#><#if(parameter.Type.EndsWith("?")){#>.Value<#}#>.ToString("s", System.Globalization.CultureInfo.InvariantCulture)));
+urlBuilder_.Append("<#=parameter.Name#>=").Append(System.Uri.EscapeDataString(<#=parameter.VariableName#><#if(parameter.Type.EndsWith("?")){#>.Value<#}#>.ToString("s", System.Globalization.CultureInfo.InvariantCulture))).Append("&");
 <#  }else if(parameter.IsArray){#>
-            foreach (var item_ in <#=parameter.VariableName#>) { url_ += string.Format("<#=parameter.Name#>={0}&", System.Uri.EscapeDataString(item_.ToString())); }
+foreach (var item_ in <#=parameter.VariableName#>) { urlBuilder_.Append("<#=parameter.Name#>=").Append(System.Uri.EscapeDataString(item_.ToString())).Append("&"); }
 <#  }else{#>
-            url_ += string.Format("<#=parameter.Name#>={0}&", System.Uri.EscapeDataString(<#=parameter.VariableName#><#if(parameter.Type.EndsWith("?")){#>.Value<#}#>.ToString()));
+urlBuilder_.Append("<#=parameter.Name#>=").Append(System.Uri.EscapeDataString(<#=parameter.VariableName#><#if(parameter.Type.EndsWith("?")){#>.Value<#}#>.ToString())).Append("&");
 <#  }#>
 <#}#>
+<#if(hasQueryParameters){#>
+        urlBuilder_.Length--;

```

Comment:
I'm trying to figure out a way to unit test the execution of the generated code. I'll let you know if I came up with something.